### PR TITLE
Add cody settings editor

### PR DESCRIPTION
**Pull request by pkukielka**
Fixes CODY-1310

## Changes

This PR adds cody settings editor with schema validation and code completion.

image
image
image

## Test plan

1. Build this PR with CODY_DIR set to [this PR branch](https://github.com/sourcegraph/cody/pull/5080).
2. Run action `Open Cody Settings Editor` (you can use `shift shift` to find it)
3. Change some Cody settings. E.g. you can try to disable autocomplete:
`"cody.autocomplete.enabled": true`
4. Hit `Ctrl + S` to trigger the config update
5. Try to use autocomplete in some other file and make sure it does not work.
6. Revert your changes and make sure autocomplete is enabled again.

### CODE REVIEW

File: src/main/kotlin/com/sourcegraph/cody/config/CodySettingsChangeListener.kt [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+import com.intellij.openapi.editor.Document
+import com.intellij.openapi.fileEditor.FileDocumentManager
+import com.intellij.openapi.fileEditor.FileDocumentManagerListener
+import com.intellij.openapi.project.Project
+import com.sourcegraph.cody.agent.CodyAgentService
+import com.sourcegraph.config.ConfigUtil
+
+class CodySettingsChangeListener(private val project: Project) : FileDocumentManagerListener {
+  override fun beforeDocumentSaving(document: Document) {
```
Comment (mkondratek) [inline src/main/kotlin/com/sourcegraph/cody/config/CodySettingsChangeListener.kt:11]: Is there any risk that the restart will happen to often while editing the file?

File: src/main/kotlin/com/sourcegraph/cody/config/CodySettingsChangeListener.kt [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    val currentFile = FileDocumentManager.getInstance().getFile(document)
+    if (currentFile?.toNioPath() == ConfigUtil.getSettingsFile(project)) {
+      CodyAgentService.getInstance(project).restartAgent(project)
+      // TODO: we should instead call `extensionConfiguration_didChange` there:
```
Comment (mkondratek) [inline src/main/kotlin/com/sourcegraph/cody/config/CodySettingsChangeListener.kt:15]: Is it an issue for Cody repo?

File: src/main/kotlin/com/sourcegraph/cody/config/actions/OpenCodySettingsEditorAction.kt [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+import kotlin.io.path.name
+
+class OpenCodySettingsEditorAction : DumbAwareEDTAction("Open Cody Settings Editor") {
+  override fun actionPerformed(e: AnActionEvent) {
+    val project =
+        e.project
+            ?: run {
+              logger.warn("Project cannot be null")
+              return
+            }
```
Comment (mkondratek) [inline src/main/kotlin/com/sourcegraph/cody/config/actions/OpenCodySettingsEditorAction.kt]: we can log it but it looks like a common practice is to simply `?: return` (see our other actions and actions in IntelliJ)

File: src/main/kotlin/com/sourcegraph/cody/config/actions/OpenCodySettingsEditorAction.kt [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+      val configName = "Cody Settings"
+      val schemaConfig =
+          UserDefinedJsonSchemaConfiguration(
+              configName,
+              JsonSchemaVersion.SCHEMA_7,
+              schemaFile.toString(),
+              false,
+              listOf(
+                  UserDefinedJsonSchemaConfiguration.Item(
```
Comment (mkondratek) [inline src/main/kotlin/com/sourcegraph/cody/config/actions/OpenCodySettingsEditorAction.kt]: can you name the params?

File: src/main/kotlin/com/sourcegraph/cody/config/actions/OpenCodySettingsEditorAction.kt [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+            }
+
+    CodyEditorUtil.showDocument(project, settingsVf)
+
+    reloadSchemaAsync(project)
+  }
+
+  private fun reloadSchemaAsync(project: Project) {
+    CodyAgentService.withAgentRestartIfNeeded(project) { agent ->
+      val settingsSchema = agent.server.extensionConfiguration_getSettingsSchema(null).get()
```
Comment (mkondratek) [inline src/main/kotlin/com/sourcegraph/cody/config/actions/OpenCodySettingsEditorAction.kt:37]: so this actions not only opens the file but also fetches the config from the agent and updates the content of the file, is that right? that makes the agent the source of truth at the moment we use this action - I wonder, shouldn't it be the client holding the truth?

**Review by mkondratek — COMMENTED** [review]
probably I missed it but how do we notify the agent about the initial settings on IDE and plugin start up?

**Comment by mkondratek** [comment]
I be just started the IDE with your changes and got this:

```
java.lang.RuntimeException: java.util.concurrent.ExecutionException: java.util.concurrent.ExecutionException: org.eclipse.lsp4j.jsonrpc.ResponseErrorException: codeActions/provide: document not found for file:///Users/mkondratek/runIdeProjects/jetbrains/.idea/.sourcegraph/cody_settings.json

Error: codeActions/provide: document not found for file:///Users/mkondratek/runIdeProjects/jetbrains/.idea/.sourcegraph/cody_settings.json
    at /Users/mkondratek/IdeaProjects/jetbrains/build/sourcegraph/agent/index.js:137245:19
    at /Users/mkondratek/IdeaProjects/jetbrains/build/sourcegraph/agent/index.js:138156:18
    at runNextTicks (node:internal/process/task_queues:60:5)
    at process.processImmediate (node:internal/timers:449:9)
    at process.topLevelDomainCallback (node:domain:160:15)
    at process.callbackTrampoline (node:internal/async_hooks:128:24)
    at async /Users/mkondratek/IdeaProjects/jetbrains/build/sourcegraph/agent/index.js:136782:44
	at com.intellij.util.ExceptionUtil.rethrow(ExceptionUtil.java:132)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils.awaitWithCheckCanceled(ProgressIndicatorUtils.java:376)
	at com.sourcegraph.cody.inspections.CodyFixHighlightPass.doCollectInformation(CodyFixHighlightPass.kt:89)
	at com.intellij.codeHighlighting.TextEditorHighlightingPass.collectInformation(TextEditorHighlightingPass.java:56)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$doRun$1(PassExecutorService.java:419)
	at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1152)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$doRun$2(PassExecutorService.java:412)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$executeProcessUnderProgress$12(CoreProgressManager.java:608)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:683)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computeUnderProgress(CoreProgressManager.java:639)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:607)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:60)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.doRun(PassExecutorService.java:411)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$run$0(PassExecutorService.java:387)
	at com.intellij.openapi.application.impl.ReadMostlyRWLock.executeByImpatientReader(ReadMostlyRWLock.java:174)
	at com.intellij.openapi.application.impl.ApplicationImpl.executeByImpatientReader(ApplicationImpl.java:213)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.run(PassExecutorService.java:385)
	at com.intellij.concurrency.JobLauncherImpl$VoidForkJoinTask$1.exec(JobLauncherImpl.java:184)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)
Caused by: java.util.concurrent.ExecutionException: java.util.concurrent.ExecutionException: org.eclipse.lsp4j.jsonrpc.ResponseErrorException: codeActions/provide: document not found for file:///Users/mkondratek/runIdeProjects/jetbrains/.idea/.sourcegraph/cody_settings.json

Error: codeActions/provide: document not found for file:///Users/mkondratek/runIdeProjects/jetbrains/.idea/.sourcegraph/cody_settings.json
    at /Users/mkondratek/IdeaProjects/jetbrains/build/sourcegraph/agent/index.js:137245:19
    at /Users/mkondratek/IdeaProjects/jetbrains/build/sourcegraph/agent/index.js:138156:18
    at runNextTicks (node:internal/process/task_queues:60:5)
    at process.processImmediate (node:internal/timers:449:9)
    at process.topLevelDomainCallback (node:domain:160:15)
    at process.callbackTrampoline (node:internal/async_hooks:128:24)
    at async /Users/mkondratek/IdeaProjects/jetbrains/build/sourcegraph/agent/index.js:136782:44
	at java.base/java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:395)
	at java.base/java.util.concurrent.CompletableFuture.get(CompletableFuture.java:2022)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils.awaitWithCheckCanceled(ProgressIndicatorUtils.java:361)
	... 21 more
Caused by: java.util.concurrent.ExecutionException: org.eclipse.lsp4j.jsonrpc.ResponseErrorException: codeActions/provide: document not found for file:///Users/mkondratek/runIdeProjects/jetbrains/.idea/.sourcegraph/cody_settings.json

Error: codeActions/provide: document not found for file:///Users/mkondratek/runIdeProjects/jetbrains/.idea/.sourcegraph/cody_settings.json
    at /Users/mkondratek/IdeaProjects/jetbrains/build/sourcegraph/agent/index.js:137245:19
    at /Users/mkondratek/IdeaProjects/jetbrains/build/sourcegraph/agent/index.js:138156:18
    at runNextTicks (node:internal/process/task_queues:60:5)
    at process.processImmediate (node:internal/timers:449:9)
    at process.topLevelDomainCallback (node:domain:160:15)
    at process.callbackTrampoline (node:internal/async_hooks:128:24)
    at async /Users/mkondratek/IdeaProjects/jetbrains/build/sourcegraph/agent/index.js:136782:44
	at java.base/java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:395)
	at java.base/java.util.concurrent.CompletableFuture.get(CompletableFuture.java:1999)
	at com.sourcegraph.cody.inspections.CodyFixHighlightPass.doCollectInformation$lambda$3(CodyFixHighlightPass.kt:78)
	at com.sourcegraph.cody.agent.CodyAgentService$Companion.withAgent$lambda$2(CodyAgentService.kt:250)
	at com.intellij.openapi.application.impl.ApplicationImpl$1.run(ApplicationImpl.java:295)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:668)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:665)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1.run(Executors.java:665)
	at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: org.eclipse.lsp4j.jsonrpc.ResponseErrorException: codeActions/provide: document not found for file:///Users/mkondratek/runIdeProjects/jetbrains/.idea/.sourcegraph/cody_settings.json

Error: codeActions/provide: document not found for file:///Users/mkondratek/runIdeProjects/jetbrains/.idea/.sourcegraph/cody_settings.json
    at /Users/mkondratek/IdeaProjects/jetbrains/build/sourcegraph/agent/index.js:137245:19
    at /Users/mkondratek/IdeaProjects/jetbrains/build/sourcegraph/agent/index.js:138156:18
    at runNextTicks (node:internal/process/task_queues:60:5)
    at process.processImmediate (node:internal/timers:449:9)
    at process.topLevelDomainCallback (node:domain:160:15)
    at process.callbackTrampoline (node:internal/async_hooks:128:24)
    at async /Users/mkondratek/IdeaProjects/jetbrains/build/sourcegraph/agent/index.js:136782:44
	at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.handleResponse(RemoteEndpoint.java:220)
	at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.consume(RemoteEndpoint.java:204)
	at org.eclipse.lsp4j.jsonrpc.TracingMessageConsumer.consume(TracingMessageConsumer.java:119)
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.handleMessage(StreamMessageProducer.java:185)
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.listen(StreamMessageProducer.java:97)
	at org.eclipse.lsp4j.jsonrpc.json.ConcurrentMessageProcessor.run(ConcurrentMessageProcessor.java:114)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	... 1 more
```

(I assume my config json did not exist initially)

File: src/main/kotlin/com/sourcegraph/cody/config/actions/OpenCodySettingsEditorAction.kt (hunk 5 above)
Comment (mkondratek) [reply src/main/kotlin/com/sourcegraph/cody/config/actions/OpenCodySettingsEditorAction.kt:37]: btw, `extensionConfiguration_getSettingsSchema` always takes `null`, why is that? 🤔

File: src/main/kotlin/com/sourcegraph/cody/config/CodySettingsChangeListener.kt (hunk 1 above)
Comment (pkukielka) [reply src/main/kotlin/com/sourcegraph/cody/config/CodySettingsChangeListener.kt:11]: I think not, this happen only on Ctrl+S, or if you switch IJ window.

File: src/main/kotlin/com/sourcegraph/cody/config/actions/OpenCodySettingsEditorAction.kt (hunk 5 above)
Comment (pkukielka) [reply src/main/kotlin/com/sourcegraph/cody/config/actions/OpenCodySettingsEditorAction.kt:37]: >  I wonder, shouldn't it be the client holding the truth?
I do not think so, and I'm not sure how that would be supposed to work?
After all it's VSC code which knows which settings are used, etc.
Keeping it in client would be artificial, and would also need to be duplicated for every client.

**Comment by pkukielka** [comment]
> probably I missed it but how do we notify the agent about the initial settings on IDE and plugin start up?

`ConfigUtil::getAgentConfiguration` which in turn uses `ConfigUtil::getCustomConfiguration` takes care of that.

**Comment by pkukielka** [comment]
> I be just started the IDE with your changes and got this:
> 
> ```
> java.lang.RuntimeException: java.util.concurrent.ExecutionException: java.util.concurrent.ExecutionException: org.eclipse.lsp4j.jsonrpc.ResponseErrorException: codeActions/provide: document not found for file:///Users/mkondratek/runIdeProjects/jetbrains/.idea/.sourcegraph/cody_settings.json
> 
> Error: codeActions/provide: document not found for file:///Users/mkondratek/runIdeProjects/jetbrains/.idea/.sourcegraph/cody_settings.json
>     at /Users/mkondratek/IdeaProjects/jetbrains/build/sourcegraph/agent/index.js:137245:19
>     at /Users/mkondratek/IdeaProjects/jetbrains/build/sourcegraph/agent/index.js:138156:18
>     at runNextTicks (node:internal/process/task_queues:60:5)
>     at process.processImmediate (node:internal/timers:449:9)
>     at process.topLevelDomainCallback (node:domain:160:15)
>     at process.callbackTrampoline (node:internal/async_hooks:128:24)
>     at async /Users/mkondratek/IdeaProjects/jetbrains/build/sourcegraph/agent/index.js:136782:44
> 	at com.intellij.util.ExceptionUtil.rethrow(ExceptionUtil.java:132)
> 	at com.intellij.openapi.progress.util.ProgressIndicatorUtils.awaitWithCheckCanceled(ProgressIndicatorUtils.java:376)
> 	at com.sourcegraph.cody.inspections.CodyFixHighlightPass.doCollectInformation(CodyFixHighlightPass.kt:89)
> 	at com.intellij.codeHighlighting.TextEditorHighlightingPass.collectInformation(TextEditorHighlightingPass.java:56)
> 	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$doRun$1(PassExecutorService.java:419)
> 	at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1152)
> 	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$doRun$2(PassExecutorService.java:412)
> 	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$executeProcessUnderProgress$12(CoreProgressManager.java:608)
> 	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:683)
> 	at com.intellij.openapi.progress.impl.CoreProgressManager.computeUnderProgress(CoreProgressManager.java:639)
> 	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:607)
> 	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:60)
> 	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.doRun(PassExecutorService.java:411)
> 	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$run$0(PassExecutorService.java:387)
> 	at com.intellij.openapi.application.impl.ReadMostlyRWLock.executeByImpatientReader(ReadMostlyRWLock.java:174)
> 	at com.intellij.openapi.application.impl.ApplicationImpl.executeByImpatientReader(ApplicationImpl.java:213)
> 	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.run(PassExecutorService.java:385)
> 	at com.intellij.concurrency.JobLauncherImpl$VoidForkJoinTask$1.exec(JobLauncherImpl.java:184)
> 	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
> 	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
> 	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
> 	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
> 	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)
> Caused by: java.util.concurrent.ExecutionException: java.util.concurrent.ExecutionException: org.eclipse.lsp4j.jsonrpc.ResponseErrorException: codeActions/provide: document not found for file:///Users/mkondratek/runIdeProjects/jetbrains/.idea/.sourcegraph/cody_settings.json
> 
> Error: codeActions/provide: document not found for file:///Users/mkondratek/runIdeProjects/jetbrains/.idea/.sourcegraph/cody_settings.json
>     at /Users/mkondratek/IdeaProjects/jetbrains/build/sourcegraph/agent/index.js:137245:19
>     at /Users/mkondratek/IdeaProjects/jetbrains/build/sourcegraph/agent/index.js:138156:18
>     at runNextTicks (node:internal/process/task_queues:60:5)
>     at process.processImmediate (node:internal/timers:449:9)
>     at process.topLevelDomainCallback (node:domain:160:15)
>     at process.callbackTrampoline (node:internal/async_hooks:128:24)
>     at async /Users/mkondratek/IdeaProjects/jetbrains/build/sourcegraph/agent/index.js:136782:44
> 	at java.base/java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:395)
> 	at java.base/java.util.concurrent.CompletableFuture.get(CompletableFuture.java:2022)
> 	at com.intellij.openapi.progress.util.ProgressIndicatorUtils.awaitWithCheckCanceled(ProgressIndicatorUtils.java:361)
> 	... 21 more
> Caused by: java.util.concurrent.ExecutionException: org.eclipse.lsp4j.jsonrpc.ResponseErrorException: codeActions/provide: document not found for file:///Users/mkondratek/runIdeProjects/jetbrains/.idea/.sourcegraph/cody_settings.json
> 
> Error: codeActions/provide: document not found for file:///Users/mkondratek/runIdeProjects/jetbrains/.idea/.sourcegraph/cody_settings.json
>     at /Users/mkondratek/IdeaProjects/jetbrains/build/sourcegraph/agent/index.js:137245:19
>     at /Users/mkondratek/IdeaProjects/jetbrains/build/sourcegraph/agent/index.js:138156:18
>     at runNextTicks (node:internal/process/task_queues:60:5)
>     at process.processImmediate (node:internal/timers:449:9)
>     at process.topLevelDomainCallback (node:domain:160:15)
>     at process.callbackTrampoline (node:internal/async_hooks:128:24)
>     at async /Users/mkondratek/IdeaProjects/jetbrains/build/sourcegraph/agent/index.js:136782:44
> 	at java.base/java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:395)
> 	at java.base/java.util.concurrent.CompletableFuture.get(CompletableFuture.java:1999)
> 	at com.sourcegraph.cody.inspections.CodyFixHighlightPass.doCollectInformation$lambda$3(CodyFixHighlightPass.kt:78)
> 	at com.sourcegraph.cody.agent.CodyAgentService$Companion.withAgent$lambda$2(CodyAgentService.kt:250)
> 	at com.intellij.openapi.application.impl.ApplicationImpl$1.run(ApplicationImpl.java:295)
> 	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
> 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
> 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
> 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
> 	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:668)
> 	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:665)
> 	at java.base/java.security.AccessController.doPrivileged(Native Method)
> 	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1.run(Executors.java:665)
> 	at java.base/java.lang.Thread.run(Thread.java:829)
> Caused by: org.eclipse.lsp4j.jsonrpc.ResponseErrorException: codeActions/provide: document not found for file:///Users/mkondratek/runIdeProjects/jetbrains/.idea/.sourcegraph/cody_settings.json
> 
> Error: codeActions/provide: document not found for file:///Users/mkondratek/runIdeProjects/jetbrains/.idea/.sourcegraph/cody_settings.json
>     at /Users/mkondratek/IdeaProjects/jetbrains/build/sourcegraph/agent/index.js:137245:19
>     at /Users/mkondratek/IdeaProjects/jetbrains/build/sourcegraph/agent/index.js:138156:18
>     at runNextTicks (node:internal/process/task_queues:60:5)
>     at process.processImmediate (node:internal/timers:449:9)
>     at process.topLevelDomainCallback (node:domain:160:15)
>     at process.callbackTrampoline (node:internal/async_hooks:128:24)
>     at async /Users/mkondratek/IdeaProjects/jetbrains/build/sourcegraph/agent/index.js:136782:44
> 	at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.handleResponse(RemoteEndpoint.java:220)
> 	at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.consume(RemoteEndpoint.java:204)
> 	at org.eclipse.lsp4j.jsonrpc.TracingMessageConsumer.consume(TracingMessageConsumer.java:119)
> 	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.handleMessage(StreamMessageProducer.java:185)
> 	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.listen(StreamMessageProducer.java:97)
> 	at org.eclipse.lsp4j.jsonrpc.json.ConcurrentMessageProcessor.run(ConcurrentMessageProcessor.java:114)
> 	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
> 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
> 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
> 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
> 	... 1 more
> ```
> 
> (I assume my config json did not exist initially)

`com.sourcegraph.cody.inspections.CodyFixHighlightPass` - looks like @RXminuS changes

File: src/main/kotlin/com/sourcegraph/cody/config/actions/OpenCodySettingsEditorAction.kt (hunk 5 above)
Comment (mkondratek) [reply src/main/kotlin/com/sourcegraph/cody/config/actions/OpenCodySettingsEditorAction.kt:37]: Ok. My main concern was about persisting the settings b/w the IDE runs. Now I can see that `ConfigUtil::getAgentConfiguration` handles the initialization properly on the startup - all good 👍

**Review by RXminuS — APPROVED** [review]

**Pull request merged**